### PR TITLE
Merge 2.10.x into master

### PR DESCRIPTION
**Pull request by gkossakowski**
Merge difficulty: moderate.

The biggest trouble was caused by 55decf733eb361c235ef1bd9039e5c6226202a96 because it overlaps with 30e2e3a78f9061ea93352427cb0ca205203041f0 Please refrain from introducing any non-mandatory changes to 2.10.x branch.

Review by @retronym

### CODE REVIEW

File: src/library/scala/util/Properties.scala [hunk 1]
```diff
@@ -131,6 +131,10 @@ private[scala] trait PropertiesTrait {
   def javaVmName            = propOrEmpty("java.vm.name")
   def javaVmVendor          = propOrEmpty("java.vm.vendor")
   def javaVmVersion         = propOrEmpty("java.vm.version")
+  // this property must remain less-well-known until 2.11
+  private def javaSpecVersion       = propOrEmpty("java.specification.version")
```
Comment (retronym) [inline src/library/scala/util/Properties.scala:135]: 06606e83dc942693dd95a30f704876c41eb4d8e5 notes:

>    For 2.10.x, javaSpecVersion is provided as a private member.
>  The active test is under `run` and the `junit` test must
>   bide its time in `pending`.
> 
>    For 2.11, the private members can be public and the app test
>    replaced with the unit test.

@som-snytt Could you please submit a PR against master after this is merged with those changes?

**Comment by retronym** [comment]
LGTM

File: src/library/scala/util/Properties.scala (hunk 1 above)
Comment (som-snytt) [inline src/library/scala/util/Properties.scala:135]: In fact, I tried it last night then noticed this wasn't actually merged yet.

**Comment by gkossakowski** [comment]
I merged the change from #2782 that fixes the test failure.

**Comment by gkossakowski** [comment]
Ok, it looks like I'll need to start over.

**Pull request closed without merging**